File: src/main/java/com/example/ImageWithTextViewHolder.java
package com.example;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import butterknife.Bind;
import butterknife.ButterKnife;
import com.example.amazingview.AmazingViewHolder;
import com.example.igor.R;
import rx.functions.Action1;

public class ImageWithTextViewHolder extends AmazingViewHolder implements Action1<String> {

    @Bind(R.id.custom_image)
    ImageView image;
    @Bind(R.id.custom_text)
    TextView text;

    public ImageWithTextViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public ImageWithTextViewHolder setImageResource(int resId) {
        image.setImageResource(resId);
        return this;
    }

    public ImageWithTextViewHolder setTextResource(int resId) {
        text.setText(resId);
        return this;
    }

    @Override
    public void call(String s) {
        text.setText(s);
    }

    @Override
    protected String onClickCallback() {
        return text.getText().toString();
    }
}

File: src/main/scala/com/example/igor/viewholders/ImageWithTextViewHolder.scala
package com.example.igor.viewholders

import android.view.View
import android.widget.{ImageView, TextView}
import com.example.igor.amazingview.AmazingViewHolder
import rx.functions.Action1

class ImageWithTextViewHolder(itemView: View) extends AmazingViewHolder(itemView) with Action1[String] {

  import com.example.igor.R

  val image: ImageView = itemView.findViewById(R.id.custom_image).asInstanceOf[ImageView]
  val text: TextView = itemView.findViewById(R.id.custom_text).asInstanceOf[TextView]

  def setImageResource(resId: Int): ImageWithTextViewHolder = {
    image.setImageResource(resId)
    this
  }

  def setTextResource(resId: Int): ImageWithTextViewHolder = {
    text.setText(resId)
    this
  }

  def call(s: String) {
    text.setText(s)
  }

  protected def onClickCallback: String = {
    text.getText.toString
  }
}